File: tests/php/Model/TestEntityDataFieldsTest.php
<?php

namespace Trulioo\SDK\Test\Model;

use Trulioo\SDK\Model\Business;
use Trulioo\SDK\Model\Communication;
use Trulioo\SDK\Model\Document;
use Trulioo\SDK\Model\DriverLicence;
use Trulioo\SDK\Model\Location;
use Trulioo\SDK\Model\NationalId;
use Trulioo\SDK\Model\Passport;
use Trulioo\SDK\Model\PersonInfo;
use PHPUnit\Framework\TestCase;
use Trulioo\SDK\Model\TestEntityDataFields;

class TestEntityDataFieldsTest extends TestCase
{
    private TestEntityDataFields $testEntityDataFields;

    public function setUp(): void
    {
        $this->testEntityDataFields = new TestEntityDataFields([]);
    }

    public function testPropertyTestEntityName()
    {
        $this->testEntityDataFields->setTestEntityName('test');
        $this->assertEquals('test', $this->testEntityDataFields->getTestEntityName());
    }

    public function testPropertyPersonInfo()
    {
        $this->testEntityDataFields->setPersonInfo(new PersonInfo([
            'first_given_name' => 'test'
        ]));
        $this->assertEquals(
            new PersonInfo([
                'first_given_name' => 'test'
            ]),
            $this->testEntityDataFields->getPersonInfo()
        );
    }

    public function testPropertyLocation()
    {
        $this->testEntityDataFields->setLocation(new Location([
            'building_number' => '10'
        ]));
        $this->assertEquals(
            new Location([
                'building_number' => '10'
            ]),
            $this->testEntityDataFields->getLocation()
        );
    }

    public function testPropertyCommunication()
    {
        $this->testEntityDataFields->setCommunication(new Communication([
            'email_address' => 'support@trulioo.com'
        ]));
        $this->assertEquals(
            new Communication([
                'email_address' => 'support@trulioo.com'
            ]),
            $this->testEntityDataFields->getCommunication()
        );
    }

    public function testPropertyDriverLicence()
    {
        $this->testEntityDataFields->setDriverLicence(new DriverLicence([
            'number' => '123'
        ]));
        $this->assertEquals(
            new DriverLicence([
                'number' => '123'
            ]),
            $this->testEntityDataFields->getDriverLicence()
        );
    }

    public function testPropertyNationalIds()
    {
        $this->testEntityDataFields->setNationalIds([
            new NationalId([
                'number' => '123'
            ])
        ]);
        $this->assertEquals(
            [
                new NationalId([
                    'number' => '123'
                ])
            ],
            $this->testEntityDataFields->getNationalIds()
        );
    }

    public function testPropertyPassport()
    {
        $this->testEntityDataFields->setPassport(new Passport([
            'mrz1' => 'test'
        ]));
        $this->assertEquals(
            new Passport([
                'mrz1' => 'test'
            ]),
            $this->testEntityDataFields->getPassport()
        );
    }

    public function testPropertyDocument()
    {
        $this->testEntityDataFields->setDocument(new Document([
            'document_front_image' => 'base64'
        ]));
        $this->assertEquals(
            new Document([
                'document_front_image' => 'base64'
            ]),
            $this->testEntityDataFields->getDocument()
        );
    }

    public function testPropertyBusiness()
    {
        $this->testEntityDataFields->setBusiness(new Business([
            'business_name' => 'test'
        ]));
        $this->assertEquals(
            new Business([
                'business_name' => 'test'
            ]),
            $this->testEntityDataFields->getBusiness()
        );
    }

    public function testPropertyCountrySpecific()
    {
        $this->testEntityDataFields->setCountrySpecific([
            'country_code' => [
                'field' => 'value'
            ]
        ]);
        $this->assertEquals(
            [
                'country_code' => [
                    'field' => 'value'
                ]
            ],
            $this->testEntityDataFields->getCountrySpecific()
        );
    }
}

File: tests/unit/java/com/trulioo/sdk/model/TestEntityDataFieldsTest.java
package com.trulioo.sdk.model;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.*;

import java.util.List;
import java.util.Arrays;
import java.util.Map;
import java.util.HashMap;

public class TestEntityDataFieldsTest {
    private TestEntityDataFields testEntityDataFields;

    @BeforeEach
    public void beforeEach() {
        testEntityDataFields = new TestEntityDataFields();
    }

    @Test
    public void testLocation() {
        Location location = new Location().city("Shibuya");
        testEntityDataFields.setLocation(location);

        assertEquals(location, testEntityDataFields.getLocation());
        assertEquals(location, new TestEntityDataFields().location(location).getLocation());
    }

    @Test
    public void testTestEntityName() {
        String testEntityName = "test";
        testEntityDataFields.setTestEntityName(testEntityName);

        assertEquals(testEntityName, testEntityDataFields.getTestEntityName());
        assertEquals(testEntityName, new TestEntityDataFields().testEntityName(testEntityName).getTestEntityName());
    }

    @Test
    public void testPersonInfo() {
        PersonInfo personInfo = new PersonInfo().firstGivenName("test");
        testEntityDataFields.setPersonInfo(personInfo);

        assertEquals(personInfo, testEntityDataFields.getPersonInfo());
        assertEquals(personInfo, new TestEntityDataFields().personInfo(personInfo).getPersonInfo());
    }

    @Test
    public void testCommunication() {
        Communication communication = new Communication().emailAddress("support@trulioo.com");
        testEntityDataFields.setCommunication(communication);

        assertEquals(communication, testEntityDataFields.getCommunication());
        assertEquals(communication, new TestEntityDataFields().communication(communication).getCommunication());
    }

    @Test
    public void testDriverLicence() {
        DriverLicence driverLicence = new DriverLicence().number("123");
        testEntityDataFields.setDriverLicence(driverLicence);

        assertEquals(driverLicence, testEntityDataFields.getDriverLicence());
        assertEquals(driverLicence, new TestEntityDataFields().driverLicence(driverLicence).getDriverLicence());
    }

    @Test
    public void testNationalIds() {
        testEntityDataFields.addNationalIdsItem(new NationalId());
        testEntityDataFields.addNationalIdsItem(new NationalId());
        List<NationalId> nationalIds = Arrays.asList(new NationalId().number("123"));
        testEntityDataFields.setNationalIds(nationalIds);

        assertEquals(nationalIds, testEntityDataFields.getNationalIds());
        assertEquals(nationalIds, new TestEntityDataFields().nationalIds(nationalIds).getNationalIds());
    }

    @Test
    public void testPassport() {
        Passport passport = new Passport().mrz1("<<test>>");
        testEntityDataFields.setPassport(passport);

        assertEquals(passport, testEntityDataFields.getPassport());
        assertEquals(passport, new TestEntityDataFields().passport(passport).getPassport());
    }

    @Test
    public void testDocument() {
        Document document = new Document().documentType("NationalId");
        testEntityDataFields.setDocument(document);

        assertEquals(document, testEntityDataFields.getDocument());
        assertEquals(document, new TestEntityDataFields().document(document).getDocument());
    }

    @Test
    public void testBusiness() {
        Business business = new Business().businessName("test");
        testEntityDataFields.setBusiness(business);

        assertEquals(business, testEntityDataFields.getBusiness());
        assertEquals(business, new TestEntityDataFields().business(business).getBusiness());
    }

    @Test
    public void testCountrySpecific() {
        testEntityDataFields.putCountrySpecificItem("a", new HashMap<String, String>());
        Map<String, Map<String, String>> countrySpecific = new HashMap<String, Map<String, String>>() {
            {
                put("country_code", new HashMap<String, String>() {
                    {
                        put("field", "value");
                    }
                });
            }
        };
        testEntityDataFields.setCountrySpecific(countrySpecific);
        testEntityDataFields.putCountrySpecificItem("b", new HashMap<String, String>());

        assertEquals(countrySpecific, testEntityDataFields.getCountrySpecific());
        assertEquals(countrySpecific, new TestEntityDataFields().countrySpecific(countrySpecific).getCountrySpecific());
    }

    @Test
    public void testEquals() {
        Location location = new Location().city("Shibuya");
        TestEntityDataFields testEntityDataFields1 = new TestEntityDataFields().location(location);

        assertEquals(testEntityDataFields1, testEntityDataFields1);
        assertEquals(testEntityDataFields1, new TestEntityDataFields().location(location));
        assertNotEquals(testEntityDataFields1, new TestEntityDataFields().location(new Location().city("different")));
        assertNotEquals(testEntityDataFields1, null);
    }

    @Test
    public void testHashcode() {
        Location location = new Location().city("Shibuya");

        assertEquals(new TestEntityDataFields().location(location).hashCode(),
                new TestEntityDataFields().location(location).hashCode());
    }

    @Test
    public void testToString() {
        Location location = new Location().city("Shibuya");

        assertEquals(new TestEntityDataFields().location(location).toString(),
                new TestEntityDataFields().location(location).toString());
    }
}
